static/app/views/dashboards/widgets/bigNumberWidget/bigNumberWidgetVisualization.tsx

```diff
@@ -37,29 +48,49 @@ export function BigNumberWidgetVisualization(props: Props) {
     return <ErrorPanel error={error} />;
   }
 
-  // Big Number widgets only show one number, so we only ever look at the first item in the Discover response
-  const datum = data?.[0];
   // TODO: Instrument getting more than one data key back as an error
+  // e.g., with data that looks like `[{'apdex()': 0.8}] this pulls out `"apdex()"` or `undefined`
+  const field = Object.keys(data?.[0] ?? {})[0];
+  const value = data?.[0]?.[field];
 
-  if (isLoading || !defined(data) || !defined(datum) || Object.keys(datum).length === 0) {
+  if (isLoading) {
```

Comment:
Should the widget also have a isError state?

Reply:
It does! It's not a boolean though, it accepts `error` as a prop. Could be `Error` or just a string.